jetty-websocket/websocket-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java

```diff
@@ -255,6 +272,24 @@ public WebSocketClient(final WebSocketContainerScope scope, EventDriverFactory e
         this.sessionFactory = sessionFactory == null ? new WebSocketSessionFactory(this) : sessionFactory;
     }
 
+    WebSocketClient(final HttpContainerScope httpContainerScope)
+    {
+        this.httpClient = httpContainerScope.getHttpClient();
```

Comment:
Looks like this implementation could just delegate to the above constructor.
```java
this(httpContainerScope, null, null, httpContainerScope.getHttpClient());
```

Reply:
I'm not a fan.
There's really 2 ways the WebSocketClient is initialized.
A standalone (hence the HttpContainerScope) and as integrated into the javax.websocket server container. (the older constructors with WebSocketContainerScope)

Reply:
I agree with @lachlan-roberts. The code is virtually the same, and I don't understand why `HttpContainerScope` has been introduced?

Reply:
I removed HttpContainerScope entirely in latest update to this PR.

Reply:
It was introduced to allow javax.websocket's ClientContainer to do what it needed.
This requirement has been eliminated, so that means the HttpContainerScope has also been eliminated.